MantidQt/CustomInterfaces/src/Reflectometry/ReflSaveTabPresenter.cpp

```diff
@@ -0,0 +1,181 @@
+#include "MantidQtCustomInterfaces/Reflectometry/ReflSaveTabPresenter.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflSaveTabView.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Run.h"
+#include "Poco/File.h"
+#include "Poco/Path.h"
+
+#include <regex>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+using namespace Mantid::API;
+
+Mantid::Kernel::Logger ReflSaveTabPresenter::g_log("ReflSaveTabPresenter");
+
+/** Constructor
+* @param view :: The view we are handling
+*/
+ReflSaveTabPresenter::ReflSaveTabPresenter(IReflSaveTabView *view)
+    : m_view(view) {
+
+  FrameworkManager::Instance();
+  saveAlgs = {"SaveReflCustomAscii", "SaveReflThreeColumnAscii",
+              "SaveANSTOAscii", "SaveILLCosmosAscii"};
+  saveExts = {".dat", ".dat", ".txt", ".mft"};
+}
+
+/** Destructor
+*/
+ReflSaveTabPresenter::~ReflSaveTabPresenter() {}
+
+void ReflSaveTabPresenter::notify(IReflSaveTabPresenter::Flag flag) {
+  switch (flag) {
+  case populateWorkspaceListFlag:
+    populateWorkspaceList();
+    suggestSaveDir();
```

Comment:
If we do `suggestSaveDir()` here, does it mean that every time that users click for instance on `Refresh` the `Save dir` is going to be overwritten with the default directory, even if users entered their own path? If so, I suggest removing this from here.